plugins/telemetry-otel/src/test/java/org/opensearch/telemetry/tracing/sampler/ProbabilisticSamplerTests.java

```diff
@@ -57,8 +64,45 @@ public void testGetSamplerWithUpdatedSamplingRatio() {
         telemetrySettings.setSamplingProbability(0.02);
 
         // Need to call getSampler() to update the value of tracerHeadSamplerSamplingRatio
-        Sampler updatedProbabilisticSampler = probabilisticSampler.getSampler();
+        Sampler updatedProbabilisticSampler = probabilisticSampler.getSampler(attributes);
         assertEquals(0.02, probabilisticSampler.getSamplingRatio(), 0.0d);
     }
 
+    public void testGetSamplerWithCustomActionSamplingRatio() {
+        Settings settings = Settings.builder().put(TRACER_EXPORTER_DELAY_SETTING.getKey(), "1s").build();
+        TelemetrySettings telemetrySettings = new TelemetrySettings(
+            Settings.EMPTY,
+            new ClusterSettings(settings, Set.of(TRACER_SAMPLER_PROBABILITY, TRACER_ENABLED_SETTING, TRACER_SAMPLER_ACTION_PROBABILITY))
+        );
+        telemetrySettings.setSamplingProbability(0.02);
+        AttributesMap attributes = AttributesMap.create(1, 100);
+
+        // Probabilistic Sampler
+        ProbabilisticSampler probabilisticSampler = new ProbabilisticSampler(telemetrySettings);
+
+        Map<String, Double> filters = new HashMap<>();
+        // Setting 100% sampling for indexing request
+        filters.put("internal:cluster/coordination/join", 1.00);
+        telemetrySettings.setActionSamplingProbability(filters);
```

Comment:
Instead of directly setting the values in the setter method can you please update the setting and see if it's working? may be another test. Or better write the integration test.

Reply:
Will add an IT in follow-up commits

Reply:
Added UT.